### PR TITLE
Tendermint Chaos Test

### PR DESCRIPTION
**Pull request by brohamgoham**
Initialized tm testnet config to make it easier to add and drop nodes

### CODE REVIEW

File: docker/.gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 */genfiles/
+*/mytestnet/
```
Comment (hansl) [inline docker/.gitignore:2]: Is this being used right now?

File: docker/e2e/chaos.sh [hunk 2]
```diff
@@ -0,0 +1,122 @@
+#!/bin/bash 
+RED='\033[0;31m'
+NC='\033[0m'
+CYAN='\033[0;36m'
+PURPLE='\033[0;35m'
+GREEN='\033[0;32m'
+
+c=0
+no_sleep () {
+    echo -e "${GREEN}RUN TENDERMINT NO SLEEP CHAOS TEST ${GREEN}"
```
Comment (hansl) [inline docker/e2e/chaos.sh:10]: I saw the other day that the `-e` argument gets printed out. Are you sure `-e` is valid? It's also not in every echo.

File: docker/e2e/chaos.sh [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    echo "${RED}STOP TENDERMINT NODE 2  $c times${RED}"
+    docker stop e2e-tendermint-2-1 
+	  sleep 3
+    echo "${PURPLE}STOP TENDERMINT NODE 0  $c times${PURPLE}"
+    docker start e2e-tendermint-2-1 
+	  sleep 3
+    echo "${NC}RANDOM TIME TEST OVER${NC}"
+
+  done
+    [[ "@" ]] && echo "options: $@"
```
Comment (hansl) [inline docker/e2e/chaos.sh:117]: Identation.

File: docker/e2e/chaos.sh [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+no_sleep
+rapid_fire
+random_time
```
Comment (hansl) [inline docker/e2e/chaos.sh:122]: Newline at the end please.

File: docker/e2e/chaos.sh [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+c=0
+no_sleep () {
+    echo -e "${GREEN}RUN TENDERMINT NO SLEEP CHAOS TEST ${GREEN}"
+    while [ $c -le 14 ]
+    do
+    echo "${RED}STOP TENDERMINT NODE 0 $c times${RED}"
+    ((c++))
+    docker stop e2e-tendermint-0-1 
+    echo "${GREEN}START TENDERMINT NODE 0 $c times${GREEN}"
+	  docker start e2e-tendermint-0-1
```
Comment (hansl) [inline docker/e2e/chaos.sh:17]: Indentation.

File: docker/e2e/chaos.sh [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+    echo "${RED}STOP TENDERMINT NODE 3  $c times${RED}"
+    docker stop e2e-tendermint-3-1 
+    echo "${GREEN}START TENDERMINT NODE 3  $c times${GREEN}"
+    docker start e2e-tendermint-3-1 
+
+    echo "${NC}NO SLEEP TEST OVER${NC}"
+
+  done
+    rapid_fire
```
Comment (hansl) [inline docker/e2e/chaos.sh:37]: This line means you are running the `rapid_fire` tests twice?

File: docker/e2e/chaos.sh [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+
+rapid_fire () {
+    echo -e "${RED}RUN TENDERMINT RAPID FIRE CHAOS ${RED}"
+    while [ $c -le 30 ]
+    do
+    echo "${RED}STOP TENDERMINT CHAOS TEST $c times${RED}"
```
Comment (hansl) [inline docker/e2e/chaos.sh:45]: Indentation inside the `while...done`

File: docker/e2e/chaos.sh [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+    echo "${CYAN}STOP TENDERMINT NODE 2  $c times${CYAN}"
+    docker stop e2e-tendermint-2-1 
+	  sleep 1
+    echo "${CYAN}Start TENDERMINT NODE 2  $c times${CYAN}"
+    docker start e2e-tendermint-2-1 
+	  sleep 1
+
+  done
+    echo "${NC}RAPID FIRE TEST OVER${NC}"
+    random_time
```
Comment (hansl) [inline docker/e2e/chaos.sh:77]: This means you are running the `random_time` test twice.

File: docker/.gitignore (hunk 1 above)
Comment (brohamgoham) [reply docker/.gitignore:2]: No

File: docker/e2e/chaos.sh [hunk 9]
```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
```
Comment (hansl) [inline docker/e2e/chaos.sh:1]: I wonder how easy using BATS (https://github.com/bats-core/bats-core) would help here to define tests, WDYT?

**Review by hansl — CHANGES_REQUESTED** [review]
Few comments.

File: docker/e2e/chaos.sh (hunk 6 above)
Comment (brohamgoham) [reply docker/e2e/chaos.sh:37]: It only runs once, when no_sleep ends and same with random_intervals test they all run once; consecutively

File: docker/e2e/chaos.sh (hunk 2 above)
Comment (brohamgoham) [reply docker/e2e/chaos.sh:10]: originally was using /n new lines to print and had the -e as backslash interpreter, forgot to remove them

File: docker/e2e/chaos.sh (hunk 9 above)
Comment (brohamgoham) [reply docker/e2e/chaos.sh:1]: I'm all for it, BATS is pretty simple, I just made snapshots script as well, still tweaking it, but currently generates ledger.xxxx.tar.gz & ledger_log.txt @hansl

File: docker/e2e/chaos.sh (hunk 6 above)
Comment (hansl) [reply docker/e2e/chaos.sh:37]: It's also called at line 120, right?.

**Review by hansl — APPROVED** [review]

**Pull request merged**